src/test/java/integrationTests/mojo/verifier/CloudHubDeploymentVerifier.java

```diff
@@ -45,39 +47,83 @@
   private static final String USERNAME_ENVIRONMENT_VARIABLE = "username";
   private static final String PASSWORD_ENVIRONMENT_VARIABLE = "password";
   private static final String ACCESS_TOKEN_KEY = "access_token";
+  private static final String APPLICATIONS = "/cloudhub/api/applications";
+  private static final String APPLICATION_NAME = "maven-plugin-cloudhub-deploy-test";
   private WebTarget target;
   private String response;
   private JSONObject responseJson;
   private IDs ids = new IDs();
   private String bearerToken;
+  private Logger log = LoggerFactory.getLogger(this.getClass());
+
+
+  public void verifyIsDeployed() throws InterruptedException {
+    loginAndGetNecessaryIds();
+    assertThat("Application was not deployed", isDeployed(), is(true));
+    deleteApplication();
+  }
+
+  public boolean isDeployed() throws InterruptedException {
```

Comment:
Wouldn't make sense to refactor isDeploy and isUndeploy for something like 

> private static final Integer SLEEP_TIME= 30000;
> 
> public boolean valdidateStatus(String status) throws InterruptedException {
>     log.info("Checking if application " + APPLICATION_NAME + " for status ["+status+"]...");
>     
>     int repeat = ATTEMPTS;
>     boolean keepValidating = true;
> 
>     while (repeat > 0 && keepValidating) {      
>       JSONObject application = getApplication(getApplications());
>       
>       if (application != null) {
>         log.info(APPLICATION_NAME + " status: " + application.getString("status"));
>         if( status.equals(application.getString("status"))){
>           keepValidating = false;
>         }
>       }
> 
>       if (keepValidating) {
>         Thread.sleep(SLEEP_TIME);
>         repeat--;
>       }
>     }
> 
>     if(!keepValidating){
>       return true;
>     }else{
>       return false;
>     }
>     
>   }

Please notice the usage of constantes the status as parameter and the fact that you before were using the repeat variable in the while but not modifying its value. 

Also the contract will be if true the status to validate has been acquired if false it didn't . 

I would also suggest to throw an exception in the event that we reach the limit number of attempts and the status was not achieved.